### DISCUSSION TITLE
Map doesn't shows while running on the chrome web browser

### DISCUSSION THREAD

**Discussion by AndhikaJeremia** [maplibre/flutter-maplibre-gl#306] [Q&A]
Hi, can anyone help? I have an error while running it on the chrome web browser on linux system. it shows Text `TargetPlatform.linux is not yet supported by the map's plugin` instead showing the map

i tried running the example it runs well
is there any setups that i need to do so it will run well on chrome web browser on linux?

**Comment by AndhikaJeremia** [comment]
> "I have an error while running it on the chrome web browser on linux system. it shows Text TargetPlatform.linux is not yet supported by the map's plugin instead showing the map"
> 
> also the map doesn't shows when i tried phone web browser. the flutter run by using this command `flutter run -d web-server --web-port port-number` and then i port forward using ngrok



> Hi, can anyone help? I have an error while running it on the chrome web browser on linux system. it shows Text `TargetPlatform.linux is not yet supported by the map's plugin` instead showing the map
> 
> i tried running the example it runs well is there any setups that i need to do so it will run well on chrome web browser on linux?


also the map doesn't shows when i tried phone web browser.
the flutter run by using this command
flutter run -d web-server --web-port port-number
and then i port forward using ngrok

**Comment by m0nac0** [comment]
Have you seen this: https://github.com/maplibre/flutter-maplibre-gl?tab=readme-ov-file#web ?

**Comment by AndhikaJeremia** [comment]
i already include it and have followed according to the instructions in the documentation

**Comment by m0nac0** [comment]
That line comes from the method channel implementation at https://github.com/maplibre/flutter-maplibre-gl/blob/d73cebfb92bc32223129c337776b1db5125c5c08/maplibre_gl_platform_interface/lib/src/method_channel_mapbox_gl.dart#L195

Method channels should only be used for iOS/Android. Are you sure you are not accidentally running the linux version of your app instead of the web version?

**Comment by AndhikaJeremia** [comment]
> That line comes from the method channel implementation at
> 
> https://github.com/maplibre/flutter-maplibre-gl/blob/d73cebfb92bc32223129c337776b1db5125c5c08/maplibre_gl_platform_interface/lib/src/method_channel_mapbox_gl.dart#L195
> 
> Method channels should only be used for iOS/Android. Are you sure you are not accidentally running the linux version of your app instead of the web version?

how to make sure that i am not accidentally running the linux version of my app, cause i when i flutter run i choose the web

**Reply by m0nac0** [reply]
you can try to run `flutter run -d web-server` in a command line in your project's directory. It will show you a URL to open in the browser.
By the way: which flutter version are you using?

**Reply by AndhikaJeremia** [reply]
> you can try to run `flutter run -d web-server` in a command line in your project's directory. It will show you a URL to open in the browser.
> 
> By the way: which flutter version are you using?

still showing the text `TargetPlatform.linux is not yet supported by the map's pluggin`
flutter 3

**Reply by m0nac0** [reply]
Which exact flutter version?

**Reply by m0nac0** [reply]
I was able to replicate the issue and I'm honestly very suprised this hasn't come up earlier. Thank you for reporting it!

**Reply by AndhikaJeremia** [reply]
> I was able to replicate the issue and I'm honestly very suprised this hasn't come up earlier. Thank you for reporting it!

Sure, also thank you for replying 😁

**Comment by AndhikaJeremia** [comment]
should i create an issue for this discussion?

**Reply by m0nac0** [reply]
I already created #309 from this discussion. Sorry, I thought GitHub would automatically show this here.

**Reply by m0nac0** [reply]
By the way: in the meantime you can use the plugin as described here https://github.com/maplibre/flutter-maplibre-gl?tab=readme-ov-file#using-the-plugin-in-your-project and web should work.

**Reply by fretif** [reply]
@m0nac0  I am still facing this issue even when I use the plugin as described in the documentation.

```
  maplibre_gl:
    git:
      url: https://github.com/maplibre/flutter-maplibre-gl.git
      ref: main
``` 

It works throw Android Studio but when I'am building the project
```
flutter build web
```
and deploy it, I am facing the issue even in the a browser running on Windows. 

Do you have an idea ?

**Reply by m0nac0** [reply]
Could you please clarify a bit when exactly it does work and does not work?
The way I understand you, the map shows in debug mode, but not in release mode?
Is this only with the chrome browser?
And do you see any errors in the browser dev tools?
